utility/src/test/java/org/oppia/android/util/math/RealExtensionsTest.kt

```diff
@@ -0,0 +1,414 @@
+
+    assertThat(result).isTrue()
+  }
+
+  @Test
+  fun testIsApproximatelyEqualTo_pointFiveAndPointSix_returnsFalse() {
+    val pointFiveReal = createIrrationalReal(0.5)
+
+    val result = pointFiveReal.isApproximatelyEqualTo(0.6)
+
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun testUnaryMinus_default_throwsException() {
+    val defaultReal = Real.getDefaultInstance()
+
+    val exception = assertThrows(IllegalStateException::class) { -defaultReal }
+
+    assertThat(exception).hasMessageThat().contains("Invalid real")
+  }
+
+  @Test
+  fun testUnaryMinus_twoInteger_returnsNegativeTwoInteger() {
+    val result = -TWO_REAL
+
+    assertThat(result).isIntegerThat().isEqualTo(-2)
+  }
+
+  @Test
+  fun testUnaryMinus_negativeTwoInteger_returnsTwoInteger() {
+    val result = -NEGATIVE_TWO_REAL
+
+    assertThat(result).isIntegerThat().isEqualTo(2)
+  }
+
+  @Test
+  fun testUnaryMinus_twoOneHalf_returnsNegativeOneHalf() {
+    val result = -ONE_HALF_REAL
+
+    assertThat(result).isRationalThat().evaluatesToDoubleThat().isWithin(1e-5).of(-0.5)
+  }
+
+  @Test
+  fun testUnaryMinus_negativeOneHalf_returnsOneHalf() {
+    val result = -NEGATIVE_ONE_HALF_REAL
+
+    assertThat(result).isRationalThat().evaluatesToDoubleThat().isWithin(1e-5).of(0.5)
+  }
+
+  @Test
+  fun testUnaryMinus_pi_returnsNegativePi() {
+    val result = -PI_REAL
+
+    assertThat(result).isIrrationalThat().isWithin(1e-5).of(-PI)
+  }
+
+  @Test
+  fun testUnaryMinus_negativePi_returnsPi() {
+    val result = -NEGATIVE_PI_REAL
+
+    assertThat(result).isIrrationalThat().isWithin(1e-5).of(PI)
+  }
+
+  @Test
+  fun testAbs_twoInteger_returnsTwoInteger() {
+    val result = abs(TWO_REAL)
+
+    assertThat(result).isIntegerThat().isEqualTo(2)
+  }
+
+  @Test
+  fun testAbs_negativeTwoInteger_returnsTwoInteger() {
+    val result = abs(NEGATIVE_TWO_REAL)
+
+    assertThat(result).isIntegerThat().isEqualTo(2)
+  }
+
+  @Test
+  fun testAbs_oneHalf_returnsOneHalf() {
+    val result = abs(ONE_HALF_REAL)
+
+    assertThat(result).isRationalThat().evaluatesToDoubleThat().isWithin(1e-5).of(0.5)
+  }
+
+  @Test
+  fun testAbs_negativeOneHalf_returnsOneHalf() {
+    val result = abs(NEGATIVE_ONE_HALF_REAL)
+
+    assertThat(result).isRationalThat().evaluatesToDoubleThat().isWithin(1e-5).of(0.5)
+  }
+
+  @Test
+  fun testAbs_pi_returnsPi() {
+    val result = abs(PI_REAL)
+
+    assertThat(result).isIrrationalThat().isWithin(1e-5).of(PI)
+  }
+
+  @Test
+  fun testAbs_negativePi_returnsPi() {
+    val result = abs(NEGATIVE_PI_REAL)
+
+    assertThat(result).isIrrationalThat().isWithin(1e-5).of(PI)
+  }
+}
+
+private fun createIntegerReal(value: Int) = Real.newBuilder().apply {
+  integer = value
+}.build()
+
+private fun createRationalReal(value: Fraction) = Real.newBuilder().apply {
+  rational = value
+}.build()
+
+private fun createIrrationalReal(value: Double) = Real.newBuilder().apply {
+  irrational = value
+}.build()
```

Comment:
What if we keep these functions static only (inside companion object)?

Reply:
That would be my preference, too, but classes can only have one companion object and I put it at the top for constants. I didn't want the helpers up there since they're more appendices. This seemed like the best alternative. Any suggestions?

Reply:
I guess, in that case let's keep it here only.